extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -89,6 +107,26 @@ export class AmpAdXOriginIframeHandler {
     this.embedStateApi_ = new SubscriptionApi(
         this.iframe, 'send-embed-state', true,
         () => this.sendEmbedInfo_(this.baseInstance_.isInViewport()));
+
+    // High-fidelity positions for scrollbound animations.
+    // Protected by 'amp-animation' experiment for now.
+    if (isExperimentOn(this.baseInstance_.win, 'amp-animation')) {
+      this.positionObserverHighFidelityApi_ = new SubscriptionApi(
+        this.iframe, MessageType.SEND_POSITIONS_HIGH_FIDELITY, true, () => {
```

Comment:
Can't this be called for each child iframe that's interested? It seems like we'd only want to register once.